### PR TITLE
Delegate HTTP request responsibilities to schema.Handler

### PR DESCRIPTION
**Pull request by parkerduckworth**
### What's being changed:

All HTTP requests should be forwarded directly from the schema manager to the handler, to decouple this from the general management of the schema, resulting in a more cleanly separated set of responsibilities for each entitiy.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: usecases/schema/handler.go [hunk 1]
```diff
@@ -102,38 +112,60 @@ func NewHandler(
...
+	return h.clusterState.AllNames()
+}
+
+func (h *Handler) NodeName() string {
+	return h.clusterState.LocalName()
 }
 
-func (m *Handler) NodeName() string {
-	return m.clusterState.LocalName()
+func (h *Handler) UpdateShardStatus(ctx context.Context,
```
Comment (redouan-rhazouani) [inline usecases/schema/handler.go:150]: I'm sure whether we should make 'UpdateShardStatus()' part of the Handler. It doesn't seem to be directly related to schema updates, and, as a result, consensus may not be necessary in this context. 

As part of interface segregation, it might be a good idea to create a new class specifically dedicated to handling this aspect

File: usecases/schema/handler.go [hunk 2]
```diff
@@ -102,38 +112,60 @@ func NewHandler(
...
+	err := h.Authorizer.Authorize(principal, "update",
+		fmt.Sprintf("schema/%s/shards/%s", class, shard))
+	if err != nil {
+		return err
+	}
+
+	return h.metaWriter.UpdateShardStatus(class, shard, status)
+}
+
+func (h *Handler) ShardsStatus(ctx context.Context,
```
Comment (redouan-rhazouani) [inline usecases/schema/handler.go:162]: I'm sure whether we should make 'ShardStatus()' part of the Handler. It doesn't seem to be directly related to schema.

File: usecases/schema/manager.go [hunk 3]
```diff
@@ -415,48 +413,3 @@ func (m *Manager) ResolveParentNodes(class, shardName string) (map[string]string
 	}
 	return name2Addr, nil
 }
-
-func (m *Manager) Nodes() []string {
```
Comment (redouan-rhazouani) [inline usecases/schema/manager.go:419]: Both `Nodes()` and `nodeName()` functions appear to be unrelated to the schema. I would recommend extracting them into a new specialized class

**Review by redouan-rhazouani — CHANGES_REQUESTED** [review]

File: usecases/schema/handler.go (hunk 1 above)
Comment (etiennedi) [reply usecases/schema/handler.go:150]: I would see it as part of the schema information that should be propagated and require consensus. Right now the side-effect of a shard being set active or inactive is isolated to a single node, but in the future this could also mean it is being offloaded to cloud storage. I assume that is something that we would want the whole cluster to have consensus on?

Comment (parkerduckworth) [reply usecases/schema/handler.go:150]: This method and `ShardsStatus` handle the `/v1/schema/{className}/shards` endpoint, so currently it is expected that the schema manger (and therefore Handler) would be responsible here.

If consensus is necessary here, it would make sense to keep this logic here. If not, I agree that a different class can handle this

**Comment by parkerduckworth** [comment]
Going to merge this PR, as unit tests rely on the `Handler`.  We can make further refactors in the future
